app/src/main/java/com/github/lookupgroup27/lookup/ui/feed/Feed.kt

```diff
@@ -25,27 +27,41 @@ import androidx.core.content.ContextCompat
 import com.github.lookupgroup27.lookup.model.feed.ProximityPostFetcher
 import com.github.lookupgroup27.lookup.model.location.LocationProviderSingleton
 import com.github.lookupgroup27.lookup.model.post.Post
+import com.github.lookupgroup27.lookup.model.profile.UserProfile
 import com.github.lookupgroup27.lookup.ui.feed.components.PostItem
 import com.github.lookupgroup27.lookup.ui.navigation.BottomNavigationMenu
 import com.github.lookupgroup27.lookup.ui.navigation.LIST_TOP_LEVEL_DESTINATION
 import com.github.lookupgroup27.lookup.ui.navigation.NavigationActions
 import com.github.lookupgroup27.lookup.ui.navigation.Route
 import com.github.lookupgroup27.lookup.ui.post.PostsViewModel
+import com.github.lookupgroup27.lookup.ui.profile.ProfileViewModel
+import com.google.firebase.auth.FirebaseAuth
 
 private const val LOCATION_PERMISSION_REQUEST_CODE = 1001
+private const val NUMBER_OF_STARS = 3
 /**
  * Main screen for displaying a feed of nearby posts. Each post includes the user's image and
  * username, and the list is updated dynamically based on the user's location.
  *
  * @param postsViewModel View Model for the posts
  * @param navigationActions Actions for navigation within the app
  */
+@SuppressLint("StateFlowValueCalledInComposition")
 @Composable
 fun FeedScreen(
     postsViewModel: PostsViewModel,
     navigationActions: NavigationActions,
-    initialNearbyPosts: List<Post>? = null // Optional parameter for testing
+    initialNearbyPosts: List<Post>? = null,
+    profileViewModel: ProfileViewModel // Optional parameter for testing
```

Comment:
It looks like the comment here is intended to describe the variable above it. Would it make sense to adjust the placement so it’s directly aligned with the variable it references?